crates/viewer/re_viewer/src/open_url.rs

```diff
@@ -10,70 +12,185 @@ pub const INTRA_RECORDING_URL_SCHEME: &str = "recording://";
+    /// * intra-recording links (typically links to an entity)
+    /// * web event listeners
+    ///
+    /// This is the highest level way of opening arbitrary URLs inside the viewer.
+    /// The only higher level way of opening URLs is `ui.ctx().open_url(...)` which will
+    /// open the URL in a browser if it's not a content URL that we can open inside the viewer.
+    pub fn open(
+        self,
+        egui_ctx: &egui::Context,
+        follow_if_http: bool,
+        select_redap_source_when_loaded: bool,
+        command_sender: &CommandSender,
+    ) {
+        re_log::debug!("Opening URL: {:?}", &self);
+
+        match self {
+            Self::IntraRecordingSelection(item) => {
                 command_sender.send_system(SystemCommand::SetSelection(item));
             }
-            Err(err) => {
-                re_log::warn!("Failed to parse selection path {selection:?}: {err}");
+
+            Self::LogDataSource(mut data_source) => {
+                if let LogDataSource::RedapDatasetPartition {
+                    select_when_loaded, ..
+                } = &mut data_source
+                {
+                    // `select_when_loaded` is not encoded in the url itself. As of writing, `DataSource::from_uri` will just always set `select_when_loaded` to `true`.
+                    // We overwrite this with the passed in value.
+                    *select_when_loaded = select_redap_source_when_loaded;
+                } else if let LogDataSource::RrdHttpUrl { follow, .. } = &mut data_source {
+                    // `follow` is not encoded in the url itself. As of writing, `DataSource::from_uri` will just always set `follow` to `false`.
+                    // We overwrite this with the passed in value.
+                    *follow = follow_if_http;
+                }
+
+                command_sender.send_system(SystemCommand::LoadDataSource(data_source));
+            }
+
+            Self::RedapCatalog(uri) => {
+                command_sender.send_system(SystemCommand::AddRedapServer(uri.origin.clone()));
+                command_sender.send_system(SystemCommand::ChangeDisplayMode(
+                    DisplayMode::RedapServer(uri.origin),
+                ));
+            }
+
+            Self::RedapEntry(uri) => {
+                command_sender.send_system(SystemCommand::AddRedapServer(uri.origin));
+                command_sender
+                    .send_system(SystemCommand::SetSelection(Item::RedapEntry(uri.entry_id)));
+            }
+
+            Self::WebEventListener(url) => {
+                handle_web_event_listener(egui_ctx, &url, command_sender);
             }
         }
-    } else if let Some(url) = url.strip_prefix(WEB_EVENT_LISTENER_SCHEME) {
-        handle_web_event_listener(_egui_ctx, url, command_sender);
-    } else {
-        return Err(());
+
+        // All of these send commands, make sure they'll be processed.
+        egui_ctx.request_repaint();
```

Comment:
The fact that the system command queue is not empty at the end of a frame should automatically trigger a request repaint, so we don't have to think of it here.